TrackletAlgorithm/TrackHandler.cc

```diff
@@ -0,0 +1,86 @@
+#include "TrackHandler.h"
+
+bool TrackHandler::compareTrack(TrackStruct& trk, TrackStruct& masterTrk, unsigned int& matchFound, unsigned int mergeCondition){
+  #pragma HLS inline
+  // compare the two tracks, masterTrack and trk
+  LOOP_CompareTrack:
+  for (unsigned int layerIndex = 0; layerIndex < 4; layerIndex++){
+    #pragma HLS unroll
+    auto masterBarrelStubIndex = ap_uint<TrackFit<NBarrelStub, NDiskStub>::kTFStubIndexSize>(masterTrk._barrelStubArray[layerIndex][0].range(kBarrelStubIndexSizeMSB, kBarrelStubIndexSizeLSB));
+    auto inputBarrelStubIndex = ap_uint<TrackFit<NBarrelStub, NDiskStub>::kTFStubIndexSize>(trk._barrelStubArray[layerIndex][0].range(kBarrelStubIndexSizeMSB, kBarrelStubIndexSizeLSB));
+    auto masterDiskStubIndex = ap_uint<TrackFit<NBarrelStub, NDiskStub>::kTFStubIndexSize>(masterTrk._diskStubArray[layerIndex][0].range(kDiskStubIndexSizeMSB, kDiskStubIndexSizeLSB));
+    auto inputDiskStubIndex = ap_uint<TrackFit<NBarrelStub, NDiskStub>::kTFStubIndexSize>(trk._diskStubArray[layerIndex][0].range(kDiskStubIndexSizeMSB, kDiskStubIndexSizeLSB));
+
+    if((masterBarrelStubIndex == inputBarrelStubIndex) && (masterBarrelStubIndex > 0)){
+      matchesFoundBarrel[layerIndex][0] = 1;
+    } else {
+      matchesFoundBarrel[layerIndex][0] = 0;
+    }
+    
+    if ((masterDiskStubIndex == inputDiskStubIndex) && (masterDiskStubIndex > 0)){
+      matchesFoundDisk[layerIndex][0] = 1;
+    } else {
+      matchesFoundDisk[layerIndex][0] = 0;
+    }
+  }
```

Comment:
To be as general as possible, I think this should be broken up into two loops, one for the barrel stubs with an upper limit of `NBarrelStub`, and one for the disk stubs with an upper limit of `NDiskStub`.

This comment also applies to other loops in this file.